docs/proposals/nap-waf.md

```diff
@@ -0,0 +1,753 @@
+- Detailed error messages for troubleshooting
+
+### NGF Integration Architecture
+
+The integration leverages NGF's existing architecture:
+
+- **Single NGF Pod**: Centralized control plane in `nginx-gateway` namespace manages all WAF operations
+- **Per-Gateway Deployment**: Each Gateway with WAF enabled gets a dedicated multi-container NGINX Pod
+- **Selective WAF Enablement**: Only Gateways configured with WAF-enabled NginxProxy resources deploy NAP v5 containers
+- **Centralized Policy Management**: NGF controllers fetch policies and distribute them to appropriate NGINX Pods via the existing Agent gRPC connection
+
+## API, Customer Driven Interfaces, and User Experience
+
+### NginxProxy Resource Extension
+
+Users enable WAF through the NginxProxy resource,
+
+```yaml
+apiVersion: gateway.nginx.org/v1alpha2
+kind: NginxProxy
+metadata:
+  name: nginx-proxy-waf
+  namespace: nginx-gateway
+spec:
+  # Simple WAF enablement (hides multi-container complexity)
+  waf:
+    enabled: true
+    # configuration tweaks optional, e.g.:
+#   kubernetes:
+#     deployment:
+#       # NGINX container with NAP module (will set to default if waf enabled is true but these values are not configured)
+#       container:
+#         image:
+#           repository: private-registry.nginx.com/nginx-gateway-fabric/nginx-plus-waf
+#           tag: "2.1.0"
+
+#       # NAP v5 required containers (will set to defaults if waf enabled is true but these values are not configured)
+#       wafContainers:
+#         enforcer:
+#           image:
+#             repository: private-registry.nginx.com/nap/waf-enforcer
+#             tag: "5.6.0"
+
+#         configManager:
+#           image:
+#             repository: private-registry.nginx.com/nap/waf-config-mgr
+#             tag: "5.6.0"
+```
+
+### WafPolicy Custom Resource
+
+Due to NAP v5's pre-compilation requirement, policies must be referenced from external sources:
+
+```yaml
+apiVersion: gateway.nginx.org/v1alpha1
+kind: WafPolicy
+metadata:
+  name: production-waf-policy
+  namespace: applications
+spec:
+  # Policy source (required due to NAP v5 pre-compilation)
+  policySource:
+    fileLocation: "s3://ngf-waf-policies/production/policy-v1.2.3.tgz"
+    authSecret:
+      name: "policy-store-credentials"
+    validation:
+      checksumVerification: true
+      # Note: Policy content validation handled by NAP v5 components
+
+    # Retry configuration for policy fetch failures
+    retryPolicy:
+      attempts: 3
+      backoff: "exponential"
+      maxDelay: "5m"
+
+    # Timeout for policy downloads
+    timeout: "30s"
+
+  # Security logging configuration for app_protect_security_log directives
+  # Multiple log configurations are supported
+  securityLogs:
+  - name: "stderr-logging"
+    # Built-in logging profile (validated against allowed values)
+    logProfile: "log_all"  # log_all, log_blocked, log_grpc_all, log_grpc_blocked, log_grpc_illegal, log_illegal
+    destination:
+      stderr: true
```

Comment:
k8s recommends to avoid booleans in CRDs when possible.

https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#primitive-types

Maybe we can follow a similar pattern that's used in the Gateway API, for things like Route filters. So you have 

```
filter:
  type: RequestMirror
  requestMirror:
     mirrorConfig...
```

With a type string for each type, and then a dedicated config for each type. Then for something like `stderr`, it's just the type but there's no associated config as of right now. But it keeps it open in the future.